src/Compilers/Core/MSBuildTaskTests/DotNetSdkTests.cs

```diff
@@ -0,0 +1,288 @@
+                $@"[{root1}sub1\]=[https://raw.githubusercontent.com/M1/*]," +
+                $@"[{root1}sub2\]=[https://raw.githubusercontent.com/M2/*]",
+                File.ReadAllText(sourceLinkJsonPath));
+
+            // SourceControlInformationFeatureSupported = false:
+            VerifyValues(
+                props: $@"
+<Project>
+  <PropertyGroup>
+    <SourceControlInformationFeatureSupported>false</SourceControlInformationFeatureSupported>
+    <DeterministicSourcePaths>true</DeterministicSourcePaths>
+  </PropertyGroup>
+  <ItemGroup>
+    <SourceRoot Include=""{root1}"" SourceLinkUrl=""https://raw.githubusercontent.com/R1/*"" />
+  </ItemGroup>
+  {sourcePackageTargets}
+  {sourceLinkPackageTargets}
+</Project>",
+                targets: new[]
+                {
+                    "CoreCompile"
+                },
+                expressions: new[]
+                {
+                    "@(SourceRoot->'%(Identity): %(MappedPath)')",
+                    "$(DeterministicSourcePaths)",
+                    "$(PathMap)"
+                },
+                expectedResults: new[]
+                {
+                    $@"{root1}: /_/",
+                    $@"{root2}: /_1/",
+                    @"true",
+                    $@"{root1}=/_/,{root2}=/_1/"
+                });
+
+            AssertEx.AssertEqualToleratingWhitespaceDifferences(
+                $@"[/_/]=[https://raw.githubusercontent.com/R1/*]," +
+                $@"[/_1/]=[https://raw.githubusercontent.com/Source/Package/*]",
+                File.ReadAllText(sourceLinkJsonPath));
+
+            // No SourceLink package:
+            VerifyValues(
+                props: $@"
+<Project>
+  <PropertyGroup>
+    <SourceControlInformationFeatureSupported>true</SourceControlInformationFeatureSupported>
+    <DeterministicSourcePaths>true</DeterministicSourcePaths>
+  </PropertyGroup>
+  <ItemGroup>
+    <SourceRoot Include=""{root1}"" SourceLinkUrl=""https://raw.githubusercontent.com/R1/*"" />
+  </ItemGroup>
+  {sourcePackageTargets}
+</Project>",
+                targets: new[]
+                {
+                    "CoreCompile"
+                },
+                expressions: new[]
+                {
+                    "@(SourceRoot->'%(Identity): %(MappedPath)')",
+                    "$(DeterministicSourcePaths)",
+                    "$(PathMap)"
+                },
+                expectedResults: new[]
+                {
+                    $@"{root1}: /_/",
+                    $@"{root2}: /_1/",
+                    @"true",
+                    $@"{root1}=/_/,{root2}=/_1/"
+                });
+
+            AssertEx.AssertEqualToleratingWhitespaceDifferences(
+                $@"[/_/]=[https://raw.githubusercontent.com/R1/*]," +
+                $@"[/_1/]=[https://raw.githubusercontent.com/Source/Package/*]",
+                File.ReadAllText(sourceLinkJsonPath));
+        }
+
+        // TODO: test no TopLevelSourceRoot, duplicate source roots
```

Comment:
Are you planning to address those TODOs in this PR? 
Otherwise, replace with a follow-up github issue. Thanks

Reply:
Just forgot to remove the comment. These are now covered.